File: vortexdbg-android/src/main/java/com/vortexdbg/linux/android/dvm/array/BaseArray.java
package com.vortexdbg.linux.android.dvm.array;

import com.vortexdbg.linux.android.dvm.Array;
import com.vortexdbg.linux.android.dvm.DvmClass;
import com.vortexdbg.linux.android.dvm.DvmObject;

abstract class BaseArray<T> extends DvmObject<T> implements Array<T> {

    BaseArray(DvmClass objectType, T value) {
        super(objectType, value);
    }

    @Override
    public String toString() {
        return String.valueOf(value);
    }

}

File: vortexdbg-android/src/main/kotlin/com/vortexdbg/linux/android/dvm/array/BaseArray.kt
package com.vortexdbg.linux.android.dvm.array

import com.vortexdbg.linux.android.dvm.Array
import com.vortexdbg.linux.android.dvm.DvmClass
import com.vortexdbg.linux.android.dvm.DvmObject

abstract class BaseArray<T>(objectType: DvmClass?, value: T) : DvmObject<T>(objectType, value), Array<T> {

    override fun toString(): String {
        return java.lang.String.valueOf(value)
    }

}
